### PR TITLE
[Parallel Execution] Get rid of the output array, extract outputs sequentially

### PR DESCRIPTION
**Pull request by gelash**
Refactor output array which was an unnecessary indirection.
Extract outputs sequentially (for now) - TODO for parallelism for much larger block sizes/concurrency level.

previously https://github.com/aptos-labs/aptos-core/pull/1347 (changed branch)

Passes tests.

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/1558)

### CODE REVIEW

File: aptos-move/parallel-executor/src/executor.rs [hunk 1]
```diff
@@ -323,25 +311,22 @@ where
...
-                        outcomes.set_result(*idx, res);
-                    }
-                })
-                .collect::<()>();
-        });
+        // TODO: for large block sizes and many cores, extract outputs in parallel.
+        let mut maybe_err = None;
+        let final_results = (0..scheduler.num_txn_to_execute())
+            .collect::<Vec<TxnIndex>>()
+            .iter()
```
Comment (zekun000) [inline aptos-move/parallel-executor/src/executor.rs:318]: using for loop seems easier than the iter, we can short circuit to return error/skip rest

Comment (gelash) [reply aptos-move/parallel-executor/src/executor.rs:318]: Done

**Comment by gelash** [comment]
ping, can I land :)

**Review by zekun000 — APPROVED** [review]

File: aptos-move/parallel-executor/src/executor.rs [hunk 2]
```diff
@@ -323,25 +311,23 @@ where
             }
         });
 
-        // Extract outputs in parallel.
+        // TODO: for large block sizes and many cores, extract outputs in parallel.
+        let mut maybe_err = None;
+        let mut final_results = Vec::new();
```
Comment (zekun000) [inline aptos-move/parallel-executor/src/executor.rs:316]: we can do with_capacity() here

Comment (gelash) [reply aptos-move/parallel-executor/src/executor.rs]: perfect!

**Pull request merged**